### DISCUSSION TITLE
transmission-daemon not starting

### DISCUSSION THREAD

**Discussion by Jaerin** [haugene/docker-transmission-openvpn#2211] [Q&A]
I've had a long time running setup using docker on my Synology NAS to run the transmission-openvpn container just fine.  Recently it stopped working.  Before I really did any troubleshooting I had the bright idea of updating DSM to 7.  Little did I know that the only thing broken at the time was the ca.cert of ghostpath that could have been fixed with a custom profile and updating the NET_ADMIN.

Unfortunately after upgrading it the container wouldn't start.  After discovering the NET_ADMIN permissions and adding that the container will start and it appears the tunnel starts fine, but transmission-daemon doesn't.

I can't seem to figure what is supposed to start the actual transmission daemon because when I connect to a bash shell on the container I can see there is a transmission-daemon service I can start, but it doesn't use the environment variables to configure it.  It uses what appears to be a default settings.json in /etc/transmission-daemon.

I can see all the data paths and config seems to be mounted the way it should, but for whatever it never actually starts the transmission-daemon.

```
   "CapAdd" : [ "NET_ADMIN" ],
   "CapDrop" : [],
   "cmd" : "dumb-init /etc/openvpn/start.sh",
   "cpu_priority" : 50,
   "enable_publish_all_ports" : false,
   "enable_restart_policy" : false,
   "enabled" : false,
   "env_variables" : [
      {
         "key" : "CREATE_TUN_DEVICE",
         "value" : "true"
      },
      {
         "key" : "PATH",
         "value" : "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
      },
      {
         "key" : "DEBIAN_FRONTEND",
         "value" : "noninteractive"
      },
      {
         "key" : "OPENVPN_USERNAME",
         "value" : "<proper_username>"
      },
      {
         "key" : "OPENVPN_PASSWORD",
         "value" : "<proper_password>"
      },
      {
         "key" : "OPENVPN_PROVIDER",
         "value" : "custom"
      },
      {
         "key" : "GLOBAL_APPLY_PERMISSIONS",
         "value" : "true"
      },
      {
         "key" : "TRANSMISSION_ALT_SPEED_DOWN",
         "value" : "50"
      },
      {
         "key" : "TRANSMISSION_ALT_SPEED_ENABLED",
         "value" : "false"
      },
      {
         "key" : "TRANSMISSION_ALT_SPEED_TIME_BEGIN",
         "value" : "540"
      },
      {
         "key" : "TRANSMISSION_ALT_SPEED_TIME_DAY",
         "value" : "127"
      },
      {
         "key" : "TRANSMISSION_ALT_SPEED_TIME_ENABLED",
         "value" : "false"
      },
      {
         "key" : "TRANSMISSION_ALT_SPEED_TIME_END",
         "value" : "1020"
      },
      {
         "key" : "TRANSMISSION_ALT_SPEED_UP",
         "value" : "50"
      },
      {
         "key" : "TRANSMISSION_BIND_ADDRESS_IPV4",
         "value" : "0.0.0.0"
      },
      {
         "key" : "TRANSMISSION_BIND_ADDRESS_IPV6",
         "value" : "::"
      },
      {
         "key" : "TRANSMISSION_BLOCKLIST_ENABLED",
         "value" : "false"
      },
      {
         "key" : "TRANSMISSION_BLOCKLIST_URL",
         "value" : "http://www.example.com/blocklist"
      },
      {
         "key" : "TRANSMISSION_CACHE_SIZE_MB",
         "value" : "4"
      },
      {
         "key" : "TRANSMISSION_DHT_ENABLED",
         "value" : "true"
      },
      {
         "key" : "TRANSMISSION_DOWNLOAD_DIR",
         "value" : "/data/complete"
      },
      {
         "key" : "TRANSMISSION_DOWNLOAD_LIMIT",
         "value" : "100"
      },
      {
         "key" : "TRANSMISSION_DOWNLOAD_LIMIT_ENABLED",
         "value" : "0"
      },
      {
         "key" : "TRANSMISSION_DOWNLOAD_QUEUE_ENABLED",
         "value" : "true"
      },
      {
         "key" : "TRANSMISSION_DOWNLOAD_QUEUE_SIZE",
         "value" : "5"
      },
      {
         "key" : "TRANSMISSION_ENCRYPTION",
         "value" : "1"
      },
      {
         "key" : "TRANSMISSION_IDLE_SEEDING_LIMIT",
         "value" : "30"
      },
      {
         "key" : "TRANSMISSION_IDLE_SEEDING_LIMIT_ENABLED",
         "value" : "false"
      },
      {
         "key" : "TRANSMISSION_INCOMPLETE_DIR",
         "value" : "/data/incomplete"
      },
      {
         "key" : "TRANSMISSION_INCOMPLETE_DIR_ENABLED",
         "value" : "true"
      },
      {
         "key" : "TRANSMISSION_LPD_ENABLED",
         "value" : "false"
      },
      {
         "key" : "TRANSMISSION_MAX_PEERS_GLOBAL",
         "value" : "200"
      },
      {
         "key" : "TRANSMISSION_MESSAGE_LEVEL",
         "value" : "2"
      },
      {
         "key" : "TRANSMISSION_PEER_CONGESTION_ALGORITHM",
         "value" : ""
      },
      {
         "key" : "TRANSMISSION_PEER_ID_TTL_HOURS",
         "value" : "6"
      },
      {
         "key" : "TRANSMISSION_PEER_LIMIT_GLOBAL",
         "value" : "200"
      },
      {
         "key" : "TRANSMISSION_PEER_LIMIT_PER_TORRENT",
         "value" : "50"
      },
      {
         "key" : "TRANSMISSION_PEER_PORT",
         "value" : "51413"
      },
      {
         "key" : "TRANSMISSION_PEER_PORT_RANDOM_HIGH",
         "value" : "65535"
      },
      {
         "key" : "TRANSMISSION_PEER_PORT_RANDOM_LOW",
         "value" : "49152"
      },
      {
         "key" : "TRANSMISSION_PEER_PORT_RANDOM_ON_START",
         "value" : "false"
      },
      {
         "key" : "TRANSMISSION_PEER_SOCKET_TOS",
         "value" : "default"
      },
      {
         "key" : "TRANSMISSION_PEX_ENABLED",
         "value" : "true"
      },
      {
         "key" : "TRANSMISSION_PORT_FORWARDING_ENABLED",
         "value" : "false"
      },
      {
         "key" : "TRANSMISSION_PREALLOCATION",
         "value" : "1"
      },
      {
         "key" : "TRANSMISSION_PREFETCH_ENABLED",
         "value" : "1"
      },
      {
         "key" : "TRANSMISSION_QUEUE_STALLED_ENABLED",
         "value" : "true"
      },
      {
         "key" : "TRANSMISSION_QUEUE_STALLED_MINUTES",
         "value" : "30"
      },
      {
         "key" : "TRANSMISSION_RATIO_LIMIT",
         "value" : "2"
      },
      {
         "key" : "TRANSMISSION_RATIO_LIMIT_ENABLED",
         "value" : "false"
      },
      {
         "key" : "TRANSMISSION_RENAME_PARTIAL_FILES",
         "value" : "true"
      },
      {
         "key" : "TRANSMISSION_RPC_AUTHENTICATION_REQUIRED",
         "value" : "false"
      },
      {
         "key" : "TRANSMISSION_RPC_BIND_ADDRESS",
         "value" : "0.0.0.0"
      },
      {
         "key" : "TRANSMISSION_RPC_ENABLED",
         "value" : "true"
      },
      {
         "key" : "TRANSMISSION_RPC_HOST_WHITELIST",
         "value" : ""
      },
      {
         "key" : "TRANSMISSION_RPC_HOST_WHITELIST_ENABLED",
         "value" : "false"
      },
      {
         "key" : "TRANSMISSION_RPC_PASSWORD",
         "value" : "password"
      },
      {
         "key" : "TRANSMISSION_RPC_PORT",
         "value" : "9091"
      },
      {
         "key" : "TRANSMISSION_RPC_URL",
         "value" : "/transmission/"
      },
      {
         "key" : "TRANSMISSION_RPC_USERNAME",
         "value" : "username"
      },
      {
         "key" : "TRANSMISSION_RPC_WHITELIST",
         "value" : "127.0.0.1"
      },
      {
         "key" : "TRANSMISSION_RPC_WHITELIST_ENABLED",
         "value" : "false"
      },
      {
         "key" : "TRANSMISSION_SCRAPE_PAUSED_TORRENTS_ENABLED",
         "value" : "true"
      },
      {
         "key" : "TRANSMISSION_SCRIPT_TORRENT_DONE_ENABLED",
         "value" : "true"
      },
      {
         "key" : "TRANSMISSION_SCRIPT_TORRENT_DONE_FILENAME",
         "value" : "/config/transmission-postprocess.sh"
      },
      {
         "key" : "TRANSMISSION_SEED_QUEUE_ENABLED",
         "value" : "false"
      },
      {
         "key" : "TRANSMISSION_SEED_QUEUE_SIZE",
         "value" : "10"
      },
      {
         "key" : "TRANSMISSION_SPEED_LIMIT_DOWN",
         "value" : "100"
      },
      {
         "key" : "TRANSMISSION_SPEED_LIMIT_DOWN_ENABLED",
         "value" : "false"
      },
      {
         "key" : "TRANSMISSION_SPEED_LIMIT_UP",
         "value" : "100"
      },
      {
         "key" : "TRANSMISSION_SPEED_LIMIT_UP_ENABLED",
         "value" : "false"
      },
      {
         "key" : "TRANSMISSION_START_ADDED_TORRENTS",
         "value" : "true"
      },
      {
         "key" : "TRANSMISSION_TRASH_ORIGINAL_TORRENT_FILES",
         "value" : "false"
      },
      {
         "key" : "TRANSMISSION_UMASK",
         "value" : "2"
      },
      {
         "key" : "TRANSMISSION_UPLOAD_LIMIT",
         "value" : "100"
      },
      {
         "key" : "TRANSMISSION_UPLOAD_LIMIT_ENABLED",
         "value" : "0"
      },
      {
         "key" : "TRANSMISSION_UPLOAD_SLOTS_PER_TORRENT",
         "value" : "14"
      },
      {
         "key" : "TRANSMISSION_UTP_ENABLED",
         "value" : "false"
      },
      {
         "key" : "TRANSMISSION_WATCH_DIR",
         "value" : "/data/watch"
      },
      {
         "key" : "TRANSMISSION_WATCH_DIR_ENABLED",
         "value" : "true"
      },
      {
         "key" : "TRANSMISSION_HOME",
         "value" : "/data/transmission-home"
      },
      {
         "key" : "TRANSMISSION_WATCH_DIR_FORCE_GENERIC",
         "value" : "false"
      },
      {
         "key" : "ENABLE_UFW",
         "value" : "false"
      },
      {
         "key" : "UFW_ALLOW_GW_NET",
         "value" : "false"
      },
      {
         "key" : "UFW_EXTRA_PORTS",
         "value" : ""
      },
      {
         "key" : "UFW_DISABLE_IPTABLES_REJECT",
         "value" : "false"
      },
      {
         "key" : "TRANSMISSION_WEB_UI",
         "value" : ""
      },
      {
         "key" : "PUID",
         "value" : "1030"
      },
      {
         "key" : "PGID",
         "value" : "100"
      },
      {
         "key" : "TRANSMISSION_WEB_HOME",
         "value" : "transmission-web-control"
      },
      {
         "key" : "DROP_DEFAULT_ROUTE",
         "value" : ""
      },
      {
         "key" : "WEBPROXY_ENABLED",
         "value" : "false"
      },
      {
         "key" : "WEBPROXY_PORT",
         "value" : "8888"
      },
      {
         "key" : "WEBPROXY_USERNAME",
         "value" : ""
      },
      {
         "key" : "WEBPROXY_PASSWORD",
         "value" : ""
      },
      {
         "key" : "HEALTH_CHECK_HOST",
         "value" : "google.com"
      },
      {
         "key" : "DOCKER_LOG",
         "value" : "true"
      },
      {
         "key" : "LOG_TO_STDOUT",
         "value" : "true"
      },
      {
         "key" : "LOCAL_NETWORK",
         "value" : "192.168.1.0/24"
      },
      {
         "key" : "OPENVPN_CONFIG",
         "value" : "default"
      }
   ],
   "exporting" : false,
   "id" : "194e7ee82cc31ad17a7e9d960aa68f952d7cb96be0a6762930809fe599ea74a9",
   "image" : "haugene/transmission-openvpn:latest",
   "is_ddsm" : false,
   "is_package" : false,
   "links" : [],
   "memory_limit" : 0,
   "name" : "haugene-transmission-openvpn1",
   "network" : [
      {
         "driver" : "bridge",
         "name" : "bridge"
      }
   ],
   "network_mode" : "bridge",
   "port_bindings" : [
      {
         "container_port" : 9091,
         "host_port" : 9091,
         "type" : "tcp"
      }
   ],
   "privileged" : false,
   "shortcut" : {
      "enable_shortcut" : false
   },
   "use_host_network" : false,
   "volume_bindings" : [
      {
         "host_volume_file" : "/docker/transmission-openvpn/United-States-Houston.ovpn",
         "mount_point" : "/etc/openvpn/custom/default.ovpn",
         "type" : "rw"
      },
      {
         "host_volume_file" : "/docker/transmission-openvpn",
         "mount_point" : "/config",
         "type" : "rw"
      },
      {
         "host_volume_file" : "/Media/Downloads/Torrents",
         "mount_point" : "/data",
         "type" : "rw"
      },
      {
         "host_volume_file" : "/docker/transmission-openvpn/resolv.conf",
         "mount_point" : "/etc/resolv.conf",
         "type" : "ro"
      }
   ]
}
```

**Comment by pkishino** [comment]
once the tunnel is up, the openvpn process will trigger this script:
https://github.com/haugene/docker-transmission-openvpn/blob/master/openvpn/tunnelUp.sh
which then starts the transmission script to start the app

**Reply by Jaerin** [reply]
That only seems to run tinyproxy.  I don't see anything in the script that then goes on to call anything from there or in the start.sh that is in the tinyproxy directory either.

**Reply by pkishino** [reply]
? this line literally calls the transmission start script:
https://github.com/haugene/docker-transmission-openvpn/blob/8cc1870cc039201e0e2f8b7684a9f4e96ae03ab9/openvpn/tunnelUp.sh#L54
which then starts the daemon here:
https://github.com/haugene/docker-transmission-openvpn/blob/8cc1870cc039201e0e2f8b7684a9f4e96ae03ab9/transmission/start.sh#L83

**Reply by Jaerin** [reply]
Strange that's not the startup script that I have.  I literally pulled this from the docker registry as a new image.

root@haugene-transmission-openvpn1:/etc/openvpn# cat tunnelUp.sh
#!/bin/bash
/etc/transmission/start.sh "$@"
[[ ! -f /opt/tinyproxy/start.sh ]] || /opt/tinyproxy/start.sh
root@haugene-transmission-openvpn1:/etc/openvpn#

**Reply by pkishino** [reply]
that would be from a very very old version.. which registry are you pulling from?

**Comment by edgd1er** [comment]
no logs, no support possible i'm afraid. 
using service in docker in a bad idea. either a container is one app start (mysql, traeffik, ...) or a supervisor handle apps start/stop. supervisor may be supervisorctl, runnit, s6-overlay ... transmission daemon is not started using services or usual systemctl commands.
In this project, there is no service manager. all apps are ran sequentially.

As for your interrogation, this line indicates how env variables are written to settings.
https://github.com/haugene/docker-transmission-openvpn/blob/8cc1870cc039201e0e2f8b7684a9f4e96ae03ab9/transmission/start.sh#L57

a bit more of set up:
https://github.com/haugene/docker-transmission-openvpn/blob/8cc1870cc039201e0e2f8b7684a9f4e96ae03ab9/transmission/start.sh#L68

then a shell is forked to start transmission:
https://github.com/haugene/docker-transmission-openvpn/blob/8cc1870cc039201e0e2f8b7684a9f4e96ae03ab9/transmission/start.sh#L82-L83

**Reply by Jaerin** [reply]
What log are you looking for?

022-03-09 06:19:14 | stdout | Wed Mar  9 06:19:14 2022 library versions: OpenSSL 1.1.1c  28 May 2019, LZO 2.10
-- | -- | --
2022-03-09 06:19:14 | stdout | Wed Mar  9 06:19:14 2022 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on May 27 2020
2022-03-09 06:19:14 | stdout | adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0
2022-03-09 06:19:14 | stdout | Setting OPENVPN credentials...
2022-03-09 06:19:14 | stdout | Starting OpenVPN using config default.ovpn
2022-03-09 06:19:14 | stdout | Using OpenVPN provider: custom

That's all that shows up in the log section of the docker container that I can find.  There doesn't seem to be an updates to the /data/transmission-home/transmission.log since before I upgraded.

**Reply by pkishino** [reply]
if that is all that is in the logs, then something is very wrong..
the logs start like this e.g.:
```
Starting container with revision: e6fd367db74075e2b507d420191a55a43b5e8d90
Creating TUN device /dev/net/tun
Using OpenVPN provider: CUSTOM
Running with VPN_CONFIG_SOURCE auto
No bundled config script found for CUSTOM. Defaulting to external config
Downloading configs from https://github.com/haugene/vpn-configs-contrib/archive/main.zip into /tmp/tmp.l0qYbFjZLo
```
and should end with following if everything is ok:
```
STARTING TRANSMISSION
Provider PIA has a script for automatic port forwarding. Will run it now.
If you want to disable this, set environment variable DISABLE_PORT_UPDATER=true
Transmission startup script complete.
```

**Reply by pkishino** [reply]
please post your complete docker log for the container from start until the "end"

**Reply by Jaerin** [reply]
Definitely have not seen anything like that in the logs.

Is there a way to verify the version of the image that I'm using?  Maybe this hasn't updated for some reason, but I just rebuilt this new try it again and it's the same way.

**Reply by Jaerin** [reply]
Where would I find the docker log that has what you are looking for?

**Reply by pkishino** [reply]
docker logs "container-name"

**Comment by Jaerin** [comment]
Tue Mar  8 08:08:33 2022 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on May 27 2020

Tue Mar  8 08:08:33 2022 library versions: OpenSSL 1.1.1c  28 May 2019, LZO 2.10

Enter Auth Username:
Using OpenVPN provider: custom

No VPN configuration provided. Using default.

Setting OPENVPN credentials...

adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0

Tue Mar  8 08:17:31 2022 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on May 27 2020

Tue Mar  8 08:17:31 2022 library versions: OpenSSL 1.1.1c  28 May 2019, LZO 2.10

Enter Auth Username:
Using OpenVPN provider: custom

No VPN configuration provided. Using default.

Setting OPENVPN credentials...

adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0

Tue Mar  8 08:45:35 2022 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on May 27 2020

Tue Mar  8 08:45:35 2022 library versions: OpenSSL 1.1.1c  28 May 2019, LZO 2.10

Enter Auth Username:
Using OpenVPN provider: custom

Starting OpenVPN using config default.ovpn

Setting OPENVPN credentials...

adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0

Tue Mar  8 17:42:59 2022 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on May 27 2020

Tue Mar  8 17:42:59 2022 library versions: OpenSSL 1.1.1c  28 May 2019, LZO 2.10

Enter Auth Username:
Using OpenVPN provider: custom

Starting OpenVPN using config default.ovpn

Setting OPENVPN credentials...

adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0

Tue Mar  8 17:45:12 2022 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on May 27 2020

Tue Mar  8 17:45:12 2022 library versions: OpenSSL 1.1.1c  28 May 2019, LZO 2.10

Enter Auth Username:
mknod: /dev/net/tun: File exists

Using OpenVPN provider: custom

Starting OpenVPN using config default.ovpn

Setting OPENVPN credentials...

adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0

Tue Mar  8 18:41:39 2022 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on May 27 2020

Tue Mar  8 18:41:39 2022 library versions: OpenSSL 1.1.1c  28 May 2019, LZO 2.10

Enter Auth Username:
mknod: /dev/net/tun: File exists

Using OpenVPN provider: custom

Starting OpenVPN using config default.ovpn

Setting OPENVPN credentials...

adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0

Options error: In [CMD-LINE]:1: Error opening configuration file: /etc/openvpn/custom/default.ovpn

Use --help for more information.

mknod: /dev/net/tun: File exists

Using OpenVPN provider: custom

Starting OpenVPN using config default.ovpn

Setting OPENVPN credentials...

adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0

Options error: In [CMD-LINE]:1: Error opening configuration file: /etc/openvpn/custom/default.ovpn

Use --help for more information.

mknod: /dev/net/tun: File exists

Using OpenVPN provider: custom

Starting OpenVPN using config default.ovpn

Setting OPENVPN credentials...

adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0

Options error: In [CMD-LINE]:1: Error opening configuration file: /etc/openvpn/custom/default.ovpn

Use --help for more information.

Using OpenVPN provider: custom

Starting OpenVPN using config default.ovpn

Setting OPENVPN credentials...

adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0

Options error: In [CMD-LINE]:1: Error opening configuration file: /etc/openvpn/custom/default.ovpn

Use --help for more information.

Using OpenVPN provider: custom

Starting OpenVPN using config default.ovpn

Setting OPENVPN credentials...

adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0

Wed Mar  9 04:00:08 2022 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on May 27 2020

Wed Mar  9 04:00:08 2022 library versions: OpenSSL 1.1.1c  28 May 2019, LZO 2.10

Enter Auth Username:
Using OpenVPN provider: custom

Starting OpenVPN using config default.ovpn

Setting OPENVPN credentials...

adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0

Wed Mar  9 04:01:18 2022 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on May 27 2020

Wed Mar  9 04:01:18 2022 library versions: OpenSSL 1.1.1c  28 May 2019, LZO 2.10

Enter Auth Username:
Using OpenVPN provider: custom

Starting OpenVPN using config default.ovpn

Setting OPENVPN credentials...

adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0

Wed Mar  9 04:06:58 2022 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on May 27 2020

Wed Mar  9 04:06:58 2022 library versions: OpenSSL 1.1.1c  28 May 2019, LZO 2.10

Enter Auth Username:
Using OpenVPN provider: custom

Starting OpenVPN using config default.ovpn

Setting OPENVPN credentials...

adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0

Wed Mar  9 06:18:22 2022 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on May 27 2020

Wed Mar  9 06:18:22 2022 library versions: OpenSSL 1.1.1c  28 May 2019, LZO 2.10

Enter Auth Username:
Using OpenVPN provider: custom

Starting OpenVPN using config default.ovpn

Setting OPENVPN credentials...

adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0

Wed Mar  9 06:19:14 2022 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on May 27 2020

Wed Mar  9 06:19:14 2022 library versions: OpenSSL 1.1.1c  28 May 2019, LZO 2.10

**Reply by pkishino** [reply]
this is the latest image
https://hub.docker.com/layers/haugene/transmission-openvpn/latest/images/sha256-6dd0c925060f042defd78753d0409fa85b1e6e130d844a2938d6f7b8bfb293da?context=explore

**Reply by pkishino** [reply]
can you check your image ID via
`` docker images`` to see if it matches.
based on your log output this is a very old image you are running..
have you done 
``
docker pull haugene/transmission-openvpn:latest
``

**Reply by Jaerin** [reply]
I apologize apparently the haugene/transmission-openvpn:latest that was showing up on my Synology was from 2020.  Probably the original image I used.  I clicked on the image in the registry and it brought me to the updated repository so I assumed it was updated.  Looking in portainter I can see it was created in 9/2020.

The docker interface in synology seems horrible, but portainer seems overly complex when configuring the container like when adding volume mapping.  I'm sure there is an easier way.

Is there a way to update a container image or easily copy the settings from one to another?

*edit* Saw your last line.  It's late I shouldn't be work on this now 😄

**Reply by pkishino** [reply]
in portainer you can easily stop the container and re-deploy and tick the box(pull latest image)

**Reply by pkishino** [reply]
manually in the command line, run the docker pull as above, then re-deploy

**Reply by edgd1er** [reply]
portainer is a good UI for container handling. Using command line `docker-compose` or `docker compose`, depending on your docker version,  is what you are looking for.

this project https://github.com/Red5d/docker-autocompose/blob/master/README.md generated a docker-compose.yml from a running container, including volumes defintion and env settings.

If you have never used docker-compose I would stick to portainer for the moment and keep docker-compose when problems will be solved.

**Comment by Jaerin** [comment]
Think I'm getting closer.  Now when I have it set to use custom and have it point to the default.ovpn that I have linked I get the following:

This is using a fresh download of the ovpn file from ghostpath just in case.  The reason I'm using it as custom is the built in ghostpath files right now have an invalid ca.cert causing problems.  The work around appeared to be specifying a custom provider and point it at the file.

Any ideas?  Thanks for all the help BTW

Starting container with revision: 8cc1870cc039201e0e2f8b7684a9f4e96ae03ab9
Creating TUN device /dev/net/tun
Using OpenVPN provider: CUSTOM
Running with VPN_CONFIG_SOURCE local
No VPN configuration provided. Using default.
Modifying /etc/openvpn/custom/default.ovpn for best behaviour in this container
Modification: Point auth-user-pass option to the username/password file
Modification: Change ca certificate path
Modification: Change ping options
Modification: Update/set resolv-retry to 15 seconds
Modification: Change tls-crypt keyfile path
Modification: Set output verbosity to 3
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
Setting OpenVPN credentials...
Wed Mar  9 07:26:36 2022 disabling NCP mode (--ncp-disable) because not in P2MP client or server mode
Options error: You must define TUN/TAP device (--dev)
Use --help for more information.

**Comment by edgd1er** [comment]
If you wish to integrate the CA certificate into the image:
`docker cp CArootCertificate.crt  /usr/local/share/ca-certificates/`
then inside the container, update certificates db: `/usr/sbin/update-ca-certificates`

At the moment, I guess you may have a problem with the /dev/tun interface, either mount it or set `CREATE_TUN_DEVICE=true`
DSM settings may vary as Synology has its own way to implement docker.

**Reply by Jaerin** [reply]
How do  you keep the container up in order to update this when it immediately shuts it down due to the error?

**Reply by Jaerin** [reply]
I was originally mounting it with a TUN.sh script

```
#!/bin/sh

# Create the necessary file structure for /dev/net/tun
if ( [ ! -c /dev/net/tun ] ); then
	if ( [ ! -d /dev/net ] ); then
		mkdir -m 755 /dev/net
	fi
	mknod /dev/net/tun c 10 200
fi

# Load the tun module if not already loaded
if ( !(lsmod | grep -q "^tun\s") ); then
	insmod /lib/modules/tun.ko

fi
```

I can see the /dev/net/tun exists on the system.

I also tried removing the task and the device then rebooted my synology.  Started the container with CREATE_TUN_DEVICE=true, but then it says it can't find the TUN device.


Start container haugene-transmission-openvpn2 failed: {"message":"error gathering device information while adding custom device \"/dev/net/tun\": no such file or directory"}.


How would I go about updating the CA certificate if I can't get the container to stay up?
Also would I just pull the CA cert section of the ovpn file to create the CArootCertificate.crt?

**Reply by edgd1er** [reply]
I guess you have tried latest tag for the image. the dev and master have diverged long time ago and dev has a lot of new features or fixes.
Could you tried that ? 
or not as this part was present in both version dev and master: https://github.com/haugene/docker-transmission-openvpn/blob/85d3b9633607f2d1fd58d21190a84a31a4cea7b8/openvpn/start.sh#L36-L43

if the dev/net/tun interface was not created, then there must be an error in the logs. running with privilege and/or net_admin capabilities could help.

**Reply by Jaerin** [reply]
Okay tried the dev version and getting some different results this time with Ghostpath.  With the CREATE_TUN_DEVICE it actually spits out a bunch of device is busy messages and then fails.  If I create the TUN device on the synology and don't use that I get a different error that I don't seem to have the log for anymore.  This all appears to be problems with Ghostpath because when I switched to using NORDVPN and setup that things worked quite well.

At this point I'd say that Ghostpath isn't supported until the certs and whatnot can be straightened out.

Thanks for the assistance.
